lastModified (#1051)
observablehq/framework: docs/javascript/files.md

### Static analysis

> The `FileAttachment` function can _only_ be passed a static string literal; constructing a dynamic path such as `FileAttachment("my" + "file.csv")` is invalid syntax. Static analysis is used to invoke [data loaders](../loaders) at build time, and ensures that only referenced files are included in the generated output during build. In the future [#260](https://github.com/observablehq/framework/issues/260), it will also allow content hashes for cache breaking during deploy.

Author: This "in the future" comment was already obsolete.

> The `FileAttachment` function can _only_ be passed a static string literal; constructing a dynamic path such as `FileAttachment("my" + "file.csv")` is invalid syntax. Static analysis is used to invoke [data loaders](../loaders) at build time, and ensures that only referenced files are included in the generated output during build. This also allows to include a content hash in the file name for cache breaking during deploy.

Reviewer: ```suggestion
The `FileAttachment` function can _only_ be passed a static string literal; constructing a dynamic path such as `FileAttachment("my" + "file.csv")` is invalid syntax. Static analysis is used to invoke [data loaders](../loaders) at build time, and ensures that only referenced files are included in the generated output during build. This also allows content hashes in the file name for cache breaking during deploy.
```

None of the files in `frames` above are loaded until a [content method](#supported-formats) is invoked, for example by saying `frames[0].image()`.

> ### Edge cases
> 

Reviewer: ```suggestion
```

> Missing files don’t exhibit a lastModified property. To determine the file’s MIME type, its extension is checked against the [mime](https://www.npmjs.com/package/mime) database; it defaults to `application/octet-stream`.

Reviewer: ```suggestion
For missing files, `file.lastModified` is undefined. The’s MIME type is determined by checking the file extension against the [`mime-db` media type database](https://github.com/jshttp/mime-db); it defaults to `application/octet-stream`.
```
